source/asdmanager.py

```diff
@@ -247,42 +233,43 @@ def _validate_and_retrieve_pre_config():
-        return
+        return {}
 
     with open(PRECONFIG_FILE) as pre_config:
         try:
             config = json.loads(pre_config.read())
         except Exception as ex:
-            print Interactive.boxed_message(['JSON contents could not be retrieved from file {0}.\nError message: {1}'.format(PRECONFIG_FILE, ex)])
+            print '\n' + Interactive.boxed_message(['JSON contents could not be retrieved from file {0}.\nError message: {1}'.format(PRECONFIG_FILE, ex)])
             sys.exit(1)
 
     if 'asdmanager' not in config or not isinstance(config['asdmanager'], dict):
-        print Interactive.boxed_message(['The ASD manager pre-configuration file must contain a "asdmanager" key with a dictionary as value'])
+        print '\n' + Interactive.boxed_message(['The ASD manager pre-configuration file must contain a "asdmanager" key with a dictionary as value'])
         sys.exit(1)
 
     errors = []
     config = config['asdmanager']
     actual_keys = config.keys()
-    expected_keys = ['api_ip', 'api_port', 'asd_ips', 'asd_start_port']
+    expected_keys = ['api_ip', 'api_port', 'asd_ips', 'asd_start_port', 'configuration_store']
     for key in actual_keys:
         if key not in expected_keys:
             errors.append('Key {0} is not supported by the ASD manager'.format(key))
     if len(errors) > 0:
-        print Interactive.boxed_message(['Errors found while verifying pre-configuration:',
-                                         ' - {0}'.format('\n - '.join(errors)),
-                                         '',
-                                         'Allowed keys:\n'
-                                         ' - {0}'.format('\n - '.join(expected_keys))])
+        print '\n' + Interactive.boxed_message(['Errors found while verifying pre-configuration:',
+                                                ' - {0}'.format('\n - '.join(errors)),
+                                                '',
+                                                'Allowed keys:\n'
+                                                ' - {0}'.format('\n - '.join(expected_keys))])
         sys.exit(1)
 
     try:
         ExtensionsToolbox.verify_required_params(actual_params=config,
                                                  required_params={'api_ip': (str, ExtensionsToolbox.regex_ip, True),
                                                                   'asd_ips': (list, ExtensionsToolbox.regex_ip, False),
                                                                   'api_port': (int, {'min': 1025, 'max': 65535}, False),
-                                                                  'asd_start_port': (int, {'min': 1025, 'max': 65435}, False)})
+                                                                  'asd_start_port': (int, {'min': 1025, 'max': 65435}, False),
+                                                                  'configuration_store': (str, ['arakoon'], False)})
     except RuntimeError as rte:
```

Comment:
Could be handy to log this exception somewhere as well.( For automation purposes?)